packages/@o3r/telemetry/src/environment/index.ts

```diff
@@ -45,48 +46,104 @@ function getPackageManagerVersion(): string | undefined {
   }
 }
 
+interface PackageManagerInfo {
+  name: string;
+  version?: string;
+}
+
 /**
  * Get package manager information
  */
-function getPackageManagerInfo() {
+function getPackageManagerInfo(): PackageManagerInfo {
   return {
     name: getPackageManager(),
     version: getPackageManagerVersion()
   };
 }
 
+interface OperatingSystemInfo {
+  architecture: string;
```

Comment:
missing comments.
The interface is not directly exported but exposed to outside via `EnvironmentMetricData `